classes/submission/Submission.inc.php

```diff
@@ -248,6 +248,42 @@ function setHideAuthor($hideAuthor) {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Return submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId() {
+		$submissionId = $this->getId();
+		$contextId = $this->getContextId();
+
+		// handle publisher ids
+		if (!ctype_digit("$submissionId")) {
```

Comment:
hm, true the function says it should return an int, but there is no check 
https://github.com/pkp/pkp-lib/blob/master/classes/core/DataObject.inc.php#L173-L175
and I tested it, if I remove the if clause articles with custom ids are not displayed... so I would guess the function also returns the custom id if there is one